pytorch_lightning/core/lightning.py

```diff
@@ -178,6 +179,24 @@ def datamodule(self) -> Any:
         )
         return self._datamodule
 
+    @property
+    def loaded_optimizer_states_dict(self) -> dict:
+        warning_cache.deprecation(
+            "The `LightningModule.loaded_optimizer_states_dict` property is deprecated in v1.4"
+            " and will be removed in v1.6.",
+            stacklevel=6,
```

Comment:
oh that's interesting. do we really need it (the stack level)? cc @carmocca

Reply:
I was following @carmocca 's example for how the datamodule property was deprecated. I could switch to using rank_zero_deprecation though.

Reply:
Yes. Otherwise it will point to the getter and not the piece or code calling the getter